769 Resolved Unit Test Guidance and Programming Strategy overlap

millerg23 opened the pull request:
Moved info from Unit Testing section in Programming Strategy Vignette into Unit Test Guidance Vignette, and replaced with link to Unit Test Guidance.

bundfussr commented on vignettes/programming_strategy.Rmd:
```diff
@@ -374,25 +374,8 @@ where the labels are stored.
-
-If a unit test fails, it must provide a helpful error message, e.g., if the output dataset is not as expected, the observations, variables, and values which differ should be provided. See also test helper function (link to be added).
+A function requires a set of unit tests to verify it produces the expected result.
+See [Writing Unit Tests in {admiral}](unit_test_guidance.html#Writing Unit Tests in {admiral}) for details.
```
The link does not work. `unit_test_guidance.html#writing-unit-tests-in-admiral-` should work.

bundfussr commented on vignettes/programming_strategy.Rmd:
```diff
@@ -374,25 +374,8 @@ where the labels are stored.
-
-The testing framework to be used is [testthat](https://testthat.r-lib.org/).
-
-Unit tests should cover the functionality of the function.
-If another function g is called within a function f, the unit tests of f should not test the functionality of g. 
-This should be tested by the unit tests of g, i.e. unit tests should be added at the lowest level.
```
I think we lost this part. Was this intended?

rossfarrugia replied:
agree - i would keep as things like the naming convention are important reminders here

bundfussr commented on vignettes/programming_strategy.Rmd:
```diff
@@ -374,25 +374,8 @@ where the labels are stored.
-If a dataset needs to be created for testing purpose, it should be done so using the function `tibble::tribble()`. 
-Make sure to align columns as well. This ensures quick code readability. 
-
-If a unit test fails, it must provide a helpful error message, e.g., if the output dataset is not as expected, the observations, variables, and values which differ should be provided. See also test helper function (link to be added).
```
I think this point got lost. Was this intended?

I would replace "See also test helper function (link to be added)." with "See `expect_dfs_equal()` for comparing datasets."

rossfarrugia reviewed (changes requested):
Agree with Stefan's requests - nothing else to add

millerg23 commented:
@bundfussr I fixed the typo in link to Unit Test Guidance in the Programming Strategy Vignette.
Also, added a section 'Test Coverage' in the Unit Test Guidance.
Do you think any further updates required, regarding `expect_dfs_equal()`

Thanks!
